DataDog/ddprof
# Remove usage of libdwfl internals

# What does this PR do?

- Remove hacky usage of dwarf module internals. Instead I use the public API to retrieve the address values.
- Remove usage of dwarf frame to retrieve register values.
Although it was handy to keep track of registers, we need to avoid using the internal structures of the libdwfl library.

# Motivation

Avoid exposing the internals of an external library.

# Additional Notes

The work around the dwarf modules is somewhat incomplete. 
The behaviours in whole host make it very likely that we would need to free part of the dwarf modules. I am not seeing an API that would enable me to do so (unless I just free the dwfl object associated to a PID). 

# How to test the change?

This change impacts two behaviours
- Retrieval of dwarf modules
This is best tested in whole host mode. 
- Update of registers 
We should make sure that when we stop unwinding the registers are correctly set and that the debug frames are relevant.
